Gems/ScriptCanvas/Code/Editor/Components/EditorGraph.cpp

```diff
@@ -1130,10 +1135,70 @@ namespace ScriptCanvasEditor
             ScriptCanvas::GraphNotificationBus::Event(GetScriptCanvasId(), &ScriptCanvas::GraphNotifications::OnConnectionComplete, connectionId);
         }
 
-
+        UpdateCorrospondingImplicitConnection(scSourceEndpoint, scTargetEndpoint);
+        
         return scConnected;
     }
 
+    void EditorGraph::UpdateCorrospondingImplicitConnection(const ScriptCanvas::Endpoint& sourceEndpoint, const ScriptCanvas::Endpoint& targetEndpoint)
+    {
+        const ScriptCanvas::Node* sourceNode = FindNode(sourceEndpoint.GetNodeId());
+        const ScriptCanvas::Node* targetNode = FindNode(targetEndpoint.GetNodeId());
```

Comment:
How are we handling situations if either (or both) the source / target endpoint node id cannot be found? There should at least be an assert here if we expect that they always exist just in case something underlying changes

Reply:
I added a check for this, thanks!

Reply:
I am not sure if I should add an assert here, I am thinking that could possibly cause the editor make an assert when a user tries to make an invalid connection or doesn't drag a connection all the way to a node if someone were to reuse this function elsewhere. There aren't any other places in this file where I found an assert being made from a node not being found from its endpoints

Reply:
Asserts are usually placed where things are expected and could be indicative of a programming/logic error. For example, you would not assert on situations where the inputs are expected to be one value or another. If its expected that there are situations where either of these nodes could not be found and the caller expects this possibility, then asserts is not the right thing to do. Its your call since you are the author of this